packages/integration-tester/src/validate.ts

```diff
@@ -0,0 +1,148 @@
+    return;
+  }
+
+  runner.group(
+    `Found ${pc.yellow(tspConfigDirs.length)} TypeSpec projects`,
+    tspConfigDirs.map((projectDir) => pc.bold(relative(dir, projectDir))).join("\n"),
+  );
+
+  let successCount = 0;
+  let failureCount = 0;
+  const failedFolders: string[] = []; // Create a processor function that handles the compilation and logging
+  const processProject = async (projectDir: string) => {
+    const result = await verifyProject(runner, projectDir, suite);
+    runner.reportTaskWithDetails(
+      result.success ? "pass" : "fail",
+      relative(dir, projectDir),
+      result.output,
+    );
+    return { dir: projectDir, result };
+  };
+
+  // Run compilations in parallel with limited concurrency
+  const results = await runWithConcurrency(tspConfigDirs, COMPILATION_CONCURRENCY, processProject);
+
+  // Count successes and failures
+  for (const { dir, result } of results) {
+    if (result.success) {
+      successCount++;
+    } else {
+      failureCount++;
+      failedFolders.push(dir);
+    }
+  }
+
+  log(`\n=== Summary ===`);
+  const passed = pc.bold(pc.green(`${successCount} passed`));
+  const failed = failureCount > 0 ? pc.bold(pc.red(`${failureCount} failed`)) : undefined;
+  log([passed, failed].filter(Boolean).join(pc.gray(" | ")), pc.gray(`(${tspConfigDirs.length})`));
+
+  if (failureCount > 0) {
+    log("\nFailed folders:");
+    failedFolders.forEach((x) => log(`  - ${relative(dir, x)}`));
+
+    // Exit with failure code to make the GitHub Actions job fail
+    process.exit(1);
+  }
+}
+
+async function findTspProjects(wd: string, pattern: string): Promise<string[]> {
+  const result = await globby(pattern, {
+    cwd: wd,
+    absolute: true,
+  });
+  return result.map((x) => dirname(x));
+}
+
+async function findTspEntrypoint(
+  directory: string,
+  suite: IntegrationTestSuite,
+): Promise<string | null> {
+  try {
+    const entries = await readdir(directory);
+
+    for (const entry of suite.entrypoints ?? ["main.tsp"]) {
+      if (entries.includes(entry)) {
+        return entry;
+      }
+    }
+
+    return null;
+  } catch (error) {
+    return null;
+  }
+}
+
+async function verifyProject(
+  runner: TaskRunner,
+  dir: string,
+  suite: IntegrationTestSuite,
+): Promise<{ success: boolean; output: string }> {
+  const entrypoint = await findTspEntrypoint(dir, suite);
+
+  if (!entrypoint) {
+    const result = {
+      success: false,
+      output: "No main.tsp or client.tsp file found in directory",
+    };
+    runner.reportTaskWithDetails("fail", dir, result.output);
+    return result;
+  }
+
+  return execTspCompile(dir, entrypoint, entrypoint === "client.tsp" ? ["--dry-run"] : []);
+}
+async function execTspCompile(
+  directory: string,
+  file: string,
+  args: string[] = [],
+): Promise<{ success: boolean; output: string }> {
+  const { failed, all } = await execa("npx", ["tsp", "compile", file, "--warn-as-error", ...args], {
```

Comment:
```suggestion
  const { failed, all } = await execa("npm", ["exec", "--no", "--", "tsp", "compile", file, "--warn-as-error", ...args], {
```

This is the safest and most reliable way to exec a package that should already be installed, without any chance of invalid arg parsing.  We have a helper for this:

https://github.com/Azure/azure-rest-api-specs/blob/48cba8646b6cb51ce77bd95a2a8c3dddca068459/.github/shared/src/exec.js#L121-L123